scripts/clean-shrinkwrap.js

```diff
@@ -15,27 +15,42 @@
 
 'use strict';
 
+const _ = require('lodash');
+const fs = require('fs');
 const path = require('path');
-const packageJSON = require('../package.json');
-const spawn = require('child_process').spawn;
-const shrinkwrapIgnore = packageJSON.shrinkwrapIgnore;
 
-console.log('Removing:', shrinkwrapIgnore.join(', '));
+const NPM_SHRINKWRAP_FILE_PATH = path.join(__dirname, '..', 'npm-shrinkwrap.json');
+const shrinkwrapFile = require(NPM_SHRINKWRAP_FILE_PATH);
+const shrinkwrapIgnore = require('../package.json').shrinkwrapIgnore;
 
 /**
- * Run an npm command
- * @param {Array} command - list of arguments
- * @returns {ChildProcess}
+ * @summary Remove dependencies from a shrinkwrap object
+ * @function
+ * @private
+ *
+ * @param {Object} object - shrinkwrap object
+ * @param {String[]} dependencies - list of dependencies
+ * @returns {Object} filtered shrinkwrap object
  */
-const npm = (command) => {
-  return spawn('npm', command, {
-    cwd: path.join(__dirname, '..'),
-    env: process.env,
-    stdio: [ process.stdin, process.stdout, process.stderr ]
-  });
+const removeDependencies = (object, dependencies) => {
+  if (_.isEmpty(object.dependencies)) {
+    return object;
+  }
+
+  object.dependencies = _.chain(object.dependencies)
+    .omitBy((dependency, name) => {
+      return _.includes(dependencies, name);
+    })
+    .mapValues((dependency) => {
+      return removeDependencies(dependency, dependencies);
+    })
+    .value();
+
+  return object;
```

Comment:
So I'm guessing (I find this function hard to read, as I'm less of a JS expert than you and @jhermsmeier ) that this recursively trawls through the dependencies tree in `object`, filtering out each dependency in `dependencies`?

Perhaps the code (especially the line `removeDependencies(dependency, dependencies)`) would be easier to read if you renamed the `dependencies` parameter to `dependency_list`, so that it's harder to confuse with `object.dependencies`, and more obvious what's just a list of strings and what's a nested object-tree? \*shrug\*